examples/GroupCharacters.jl

```diff
@@ -0,0 +1,348 @@
+    end
+
+    if p != 0
+      isprime(p) || error("p must be 0 or a prime integer")
+      tbl = mod(tbl, p)
+    end
+
+    return GAPGroupCharacterTable(G, tbl, p)
+end
+
+function character_table(id::String, p::Int = 0)
+    if GAP.Globals.IsBoundGlobal(GAP.julia_to_gap("CTblLib"))
+      tbl = GAP.Globals.CharacterTable(GAP.julia_to_gap(id))
+      if tbl == GAP.Globals.fail
+        return nothing
+      elseif p != 0
+        isprime(p) || error("p must be 0 or a prime integer")
+        tbl = GAP.Globals.mod(tbl, GAP.julia_to_gap(p))
+        if tbl == GAP.Globals.fail
+          return nothing
+        end
+      end
+      return GAPGroupCharacterTable(tbl, p)
+    else
+      error("no character table library available")
+    end
+end
+
+function Base.show(io::IO, tbl::GAPGroupCharacterTable)
+    print(io, GAP.CSTR_STRING(GAP.Globals.StringDisplayObj(tbl.GAPTable)))
+end
+
+function Base.print(io::IO, tbl::GAPGroupCharacterTable)
+    gaptbl = tbl.GAPTable
+    if GAP.Globals.HasUnderlyingGroup(gaptbl)
+      id = string(GAP.Globals.UnderlyingGroup(gaptbl))
+    else
+      id = "\"" * String(GAP.Globals.Identifier(gaptbl)) * "\""
+    end
+    print(io, "character_table($id)")
+end
+
+Base.length(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
+AbstractAlgebra.nrows(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
+AbstractAlgebra.ncols(tbl::GAPGroupCharacterTable) = return length(GAP.Globals.Irr(tbl.GAPTable))
+
+function Base.getindex(tbl::GAPGroupCharacterTable, i::Int)
+    return group_class_function(tbl, GAP.Globals.Irr(tbl.GAPTable)[i])
+end
+
+function Base.getindex(tbl::GAPGroupCharacterTable, i::Int, j::Int)
+    val = GAP.Globals.Irr(tbl.GAPTable)[i, j]
+    return QabModule.QabElem(val)
+end
+
+function Base.mod(tbl::GAPGroupCharacterTable, p::Int)
+    isprime(p) || error("p must be a prime integer")
+    tbl.characteristic == 0 || error("tbl mod p only for ordinary table tbl")
+
+    modtbls = AbstractAlgebra.get_special(tbl, :brauer_tables)
+    if modtbls == nothing
+      modtbls = Dict{Int,Any}()
+      AbstractAlgebra.set_special(tbl, :brauer_tables => modtbls)
+    end
+    if ! haskey(modtbls, p)
+      modtblgap = mod(tbl.GAPTable, p)
+      if modtblgap == GAP.Globals.fail
+        modtbls[p] = nothing
+      elseif isdefined(tbl, :GAPGroup)
+        modtbls[p] = GAPGroupCharacterTable(tbl.GAPGroup, modtblgap, p)
+      else
+        modtbls[p] = GAPGroupCharacterTable(modtblgap, p)
+      end
+    end
+
+    return modtbls[p]
+end
+
+function decomposition_matrix(tbl::GAPGroupCharacterTable)
+    isprime(tbl.characteristic) || error("characteristic of tbl must be a prime integer")
+    return Matrix{Int}(GAP.Globals.DecompositionMatrix(tbl.GAPTable))
+end
+
+
+#############################################################################
+##
+##  characters
+##
+abstract type GroupClassFunction end
```

Comment:
Is there any other notion of "class function" that we might clash with, and that would justify adding this "Group" prefix? Google and Wikipedia inform me that there is a notion of "class function" in abstract set theory (a function whose domain is not a set but a proper class), but it seems unlikely this will ever matter to us?

Reply:
From my point of view, class functions are a generalization of (generalized) characters, and the term character occurs in several areas. (For example, there are character tables of endomorphism rings.) Thus it is appropriate to talk about group character objects in the given context, and then would be strange to coin the general term class function.